sklearn/manifold/_t_sne.py

```diff
@@ -597,6 +598,14 @@ class TSNE(BaseEstimator):
 
         .. versionadded:: 0.22
 
+    square_distance : {True, False, or 'legacy'}, default='legacy'
```

Comment:
The standard way to do something like this is an [enum](https://docs.python.org/3/library/enum.html). 

Although Python allows arguments to be of arbitrary types, it's usually cleaner to accept only one type (or "interface," e.g. "any iterable," i.e. "anything that implements `__iter__` and `__next__`").

Reply:
We traditionally prefer not to have an enum here because the enum would have to be scoped to this specific parameter for this specific estimator. From a CS point of view, it would be cleaner to have a single type. But for our users with non-CS backgrounds, reducing the number of abstractions, especially in the public interface, makes sklearn easy to use.

Reply:
If I understand correctly, scitkit-learn requires input validation to be done in `.fit()` specifically, rather than enforcing it in the constructor. I'm trying to find the source where I read this... [this section](https://scikit-learn.org/dev/developers/develop.html#instantiation) is the closest I've come, but I don't think it's exactly it.

Reply:
Yes, validation is done in `fit`.

Reply:
Oops, sorry, I hadn't refreshed the page. I missed all of your comments when I made mine. So sorry, @thomasjpfan.